Rename buffers menu to Window and move it.
turn off key binding hint caching because it is buggy

*** lisp/menu-bar.el	06 Apr 2008 12:51:37 +0100	1.302.2.11
--- lisp/menu-bar.el	25 Apr 2008 09:45:51 +0100	
***************
*** 50,65 ****
  ;; Force Help item to come last, after the major mode's own items.
  ;; The symbol used to be called `help', but that gets confused with the
  ;; help key.
! (setq menu-bar-final-items '(help-menu))
  
  (define-key global-map [menu-bar help-menu] (cons "Help" menu-bar-help-menu))
- (defvar menu-bar-tools-menu (make-sparse-keymap "Tools"))
- (define-key global-map [menu-bar tools] (cons "Tools" menu-bar-tools-menu))
  ;; This definition is just to show what this looks like.
  ;; It gets modified in place when menu-bar-update-buffers is called.
  (defvar global-buffers-menu-map (make-sparse-keymap "Buffers"))
  (define-key global-map [menu-bar buffer]
!   (cons "Buffers" global-buffers-menu-map))
  (defvar menu-bar-options-menu (make-sparse-keymap "Options"))
  (define-key global-map [menu-bar options]
    (cons "Options" menu-bar-options-menu))
--- 50,65 ----
  ;; Force Help item to come last, after the major mode's own items.
  ;; The symbol used to be called `help', but that gets confused with the
  ;; help key.
! (setq menu-bar-final-items '(buffer help-menu))
  
  (define-key global-map [menu-bar help-menu] (cons "Help" menu-bar-help-menu))
  ;; This definition is just to show what this looks like.
  ;; It gets modified in place when menu-bar-update-buffers is called.
  (defvar global-buffers-menu-map (make-sparse-keymap "Buffers"))
  (define-key global-map [menu-bar buffer]
!   (cons "Window" global-buffers-menu-map))
! (defvar menu-bar-tools-menu (make-sparse-keymap "Tools"))
! (define-key global-map [menu-bar tools] (cons "Tools" menu-bar-tools-menu))
  (defvar menu-bar-options-menu (make-sparse-keymap "Options"))
  (define-key global-map [menu-bar options]
    (cons "Options" menu-bar-options-menu))
*** xxx!!src/keyboard.c	25 Feb 2008 16:04:54 -0000	1.899.2.11
--- xxx!!src/keyboard.c	25 Oct 2008 16:26:30 -0000
***************
*** 7686,7693 ****
    if (NILP (cachelist))
      {
        /* We have to create a cachelist.  */
!       CHECK_IMPURE (start);
!       XSETCDR (start, Fcons (Fcons (Qnil, Qnil), XCDR (start)));
        cachelist = XCAR (XCDR (start));
        newcache = 1;
        tem = AREF (item_properties, ITEM_PROPERTY_KEYEQ);
--- 7718,7728 ----
    if (NILP (cachelist))
      {
        /* We have to create a cachelist.  */
!       /* With the introduction of where_is_cache, the computation
!          of equivalent key bindings is sufficiently fast that we
!          do not need to cache it here any more. */
!       /* CHECK_IMPURE (start);
!          XSETCDR (start, Fcons (Fcons (Qnil, Qnil), XCDR (start))); */
        cachelist = XCAR (XCDR (start));
        newcache = 1;
        tem = AREF (item_properties, ITEM_PROPERTY_KEYEQ);
